.github/workflows/release.yml

```diff
@@ -0,0 +1,33 @@
+name: Build & Release
+
+on:
+  release:
+    types: [published]
+
+permissions:
+  contents: read
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    environment:
+      name: release
+      url: https://pypi.org/p/prettyconf
+    permissions:
+      id-token: write
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip build
+
+      - name: Build package
+        run: python -m build
+
+      - name: Publish package
+        uses: pypa/gh-action-pypi-publish@v1.12.4
```

Comment:
This will require authorization on PyPI. After that, releasing will be as simple as creating a release here on Github

Reply:
What kind of auth? Can you point me to the proper documentation?

Reply:
There should be a Publishing area on the project settings on PyPI, where you should add this repository settings. 
```
Owner: osantana
Repository name: prettyconf
Workflow name: release.yml
Environment name: release
```

Then, on this project settings on Github, there should be an "Environments" area where you should create a `release` environment. Simply creating it should work, without any additional setting.